app/api/ada/lib/storage/database/genericBip44/api.js

```diff
@@ -0,0 +1,373 @@
+// @flow
+
+import type {
+  lf$Database,
+} from 'lovefield';
+
+import type {
+  Bip44DerivationMappingInsert, Bip44DerivationMappingRow,
+  Bip44DerivationInsert,
+  Bip44DerivationRow,
+  Bip44RootInsert, Bip44RootRow,
+  Bip44PurposeInsert, Bip44PurposeRow,
+  Bip44CoinTypeInsert, Bip44CoinTypeRow,
+  Bip44AccountInsert, Bip44AccountRow,
+  Bip44ChainInsert, Bip44ChainRow,
+  Bip44AddressInsert, Bip44AddressRow,
+  PublicDeriverInsert, PublicDeriverRow,
+  Bip44WrapperInsert, Bip44WrapperRow,
+} from './tables';
+import * as Tables from './tables';
+import { addToTable } from '../utils';
+import type { AddRowRequest } from '../utils';
+
+/**
+ * A specific number means you only care about the specific index
+ * Null indicates querying all derivations at the level
+ */
+export type BIP32QueryPath = Array<number | null>;
+type PathMapType = Map<number, Array<number>>;
+
+export const DerivationLevels = Object.freeze({
+  ROOT: {
+    level: 0,
+    table: Tables.Bip44RootSchema,
+  },
+  PURPOSE: {
+    level: 1,
+    table: Tables.Bip44PurposeSchema,
+  },
+  COIN_TYPE: {
+    level: 2,
+    table: Tables.Bip44CoinTypeSchema,
+  },
+  ACCOUNT: {
+    level: 3,
+    table: Tables.Bip44AccountSchema,
+  },
+  CHAIN: {
```

Comment:
I've seen that you refer to this level as **CHAIN** in the code, diagram, and specs, whereas Bip44 name this level as **CHANGE**. Is there a reason for this?

Reply:
Yes. Change is a terrible name and the Bip44 spec itself even uses both `chain` and `change` depending on the location

Reply:
Additionally the term "Change" doesn't even make sense if you consider the ad-hoc chain scenario because whether or not something is "change" depends on its index with respect to its parent Account. If you have no parent account, they're just arbitrary chains.